extensions/azure/cosmos/contract-negotiation-store-cosmos/src/test/java/org/eclipse/dataspaceconnector/contract/negotiation/store/CosmosContractNegotiationStoreIntegrationTest.java

```diff
@@ -316,43 +297,37 @@ void nextForState_leasedBySelf() {
 
         // let's verify that the first invocation correctly sets the lease
         var result = store.nextForState(state.code(), 10);
-        assertThat(result).hasSize(1); //should contain the lease already
+        assertThat(result).hasSize(1).extracting(ContractNegotiation::getId).containsExactly(n.getId()); //should contain the lease already
         var storedNegotiation = readItem(n.getId());
         assertThat(storedNegotiation.getLease()).isNotNull().hasFieldOrPropertyWithValue("leasedBy", CONNECTOR_ID);
 
         // verify that the subsequent call to nextForState does not return the entity
         result = store.nextForState(state.code(), 10);
-
         assertThat(result).isEmpty();
     }
 
     @Test
-    void nextForState_leaseByAnotherExpired() throws InterruptedException {
+    void nextForState_leaseByAnotherExpired() {
         var state = ContractNegotiationStates.CONFIRMED;
         var n = generateNegotiation(state);
         var doc = new ContractNegotiationDocument(n, partitionKey);
-        doc.acquireLease("another-connector", Duration.ofMillis(10));
-        container.createItem(doc);
-
-        Thread.sleep(20); //give the lease time to expire
-
-        var result = store.nextForState(state.code(), 10);
-        assertThat(result).hasSize(1).allSatisfy(neg -> assertThat(neg).usingRecursiveComparison().isEqualTo(n));
-    }
-
-    @Test
-    void nextForState_lockEntity() {
-        var n = generateNegotiation("test-id-lock", ContractNegotiationStates.CONSUMER_OFFERED);
-        var doc = new ContractNegotiationDocument(n, partitionKey);
+        Duration leaseDuration = Duration.ofSeconds(5);
+        doc.acquireLease("another-connector", leaseDuration);
         container.createItem(doc);
 
-        // verify nextForState sets the lease
-        var result = store.nextForState(ContractNegotiationStates.CONSUMER_OFFERED.code(), 5);
-        assertThat(result).hasSize(1).extracting(ContractNegotiation::getId).containsExactly(n.getId());
-        var storedDoc = readItem(n.getId());
+        // before the lease expired
+        var negotiationsBeforeLeaseExpired = store.nextForState(state.code(), 10);
+        assertThat(negotiationsBeforeLeaseExpired).isEmpty();
 
-        // verify subsequent call to nextforState does not return the CN
-        assertThat(store.nextForState(ContractNegotiationStates.CONSUMER_OFFERED.code(), 5)).isEmpty();
+        // after the lease expired
+        await()
+                .atMost(Duration.ofSeconds(20))
+                .pollInterval(Duration.ofSeconds(1))
+                .pollDelay(leaseDuration) //give the lease time to expire
+                .untilAsserted(() -> {
+                    List<ContractNegotiation> negotiationsAfterLeaseExpired = store.nextForState(state.code(), 10);
+                    assertThat(negotiationsAfterLeaseExpired).hasSize(1).allSatisfy(neg -> assertThat(neg).usingRecursiveComparison().isEqualTo(n));
```

Comment:
just curious here we do recursive comparison on object but in above test `nextForState_leasedBySelf` we only match  id. why is this difference ? should we do the recursive comparison at both places or the other way ?

Reply:
In `nextForState_leasedBySelf` we want to just check the id because in the next line we read the item by id:
```java
assertThat(result).hasSize(1).extracting(ContractNegotiation::getId).containsExactly(n.getId()); 
var storedNegotiation = readItem(n.getId());
```
so it's not the main assertion in the test.
And here in the `nextForState_leasedByAnotherExpired` we are aserting the whole item.

But in general it doesn't make a big difference, so I'll revert the change from `nextForState_leasedBySelf` to avoid confusion